internal/mode/static/state/dataplane/types.go

```diff
@@ -249,3 +252,19 @@ type VerifyTLS struct {
 	Hostname     string
 	RootCAPath   string
 }
+
+// Telemetry represents Otel configuration for the dataplane.
+type Telemetry struct {
+	// Endpoint specifies the address of OTLP/gRPC endpoint that will accept telemetry data.
+	Endpoint string
+	// ServiceName is the “service.name” attribute of the OTel resource.
+	ServiceName string
+	// Interval specifies the export interval.
+	Interval string
+	// SpanAttributes are custom key/value attributes that are added to each span.
+	SpanAttributes []ngfAPI.SpanAttribute
```

Comment:
I recommend we don't include API types in configuration package, so that we keep layers separated.
We also did this https://github.com/nginxinc/nginx-gateway-fabric/pull/976 :)

Reply:
What's the difference in pulling in an internal type, like `resolver.Endpoint` versus our own API type `ngfAPI.SpanAttribute`? The cons listed in that ticket don't stick out to me as cons in this case.

Reply:
our APIs are much more similar to Gateway APIs then resolver.Endpoint.
> The cons listed in that ticket don't stick out to me as cons in this case.
One more thing here is consistency. If we introduce our APIs here, I think somebody will just add Gateway APIs again later because NGF APIs are already there...

Reply:
I will say that this is going to become more difficult and tedious with our own API. Our APIs are nginx-specific, and it is difficult to come up with dataplane-agnostic terms for everything. I tried to address this in the client settings PR by adding a Customization type that holds config as a byte array, but I'm not sure how dataplane-agnostic that really is.

Reply:
Yeah with my change I'm literally just duplicating the struct. Becomes a pain when we change something in one place, it won't propagate.

Reply:
Made the change, but still open for debate in this conversation.